spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -1,69 +1,108 @@
 require "rails_helper"
 
 RSpec.describe Broadcasts::WelcomeNotification::Generator, type: :service do
+  let(:mascot_account)             { create(:user) }
+  let!(:welcome_thread)            { create(:article, user: mascot_account, published: true, tags: "welcome") }
+  let!(:welcome_broadcast)         { create(:welcome_broadcast, :active) }
+  let!(:twitter_connect_broadcast) { create(:twitter_connect_broadcast, active: true) }
+  let!(:github_connect_broadcast)  { create(:github_connect_broadcast, active: true) }
+
+  before do
+    allow(User).to receive(:mascot_account).and_return(mascot_account)
+    SiteConfig.staff_user_id = mascot_account.id
+  end
+
+  after do
+    # SiteConfig.clear_cache should work here but for some reason it isn't
+    SiteConfig.staff_user_id = 1
+  end
+
   describe "::call" do
-    let(:receiving_user) { create(:user) }
+    it "send all forms of welcome notifications" do
+      user = create(:user, :with_identity, identities: ["github"], created_at: 1.week.ago)
+      sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
+      expect(user.notifications.count).to eq(2)
+    end
```

Comment:
is this test non-specific to any welcome notification? curious why it isn't nested and why it has a github identity

Reply:
I should've left a comment. This spec is incomplete and will be modified with each new welcome notifications. The class method `::call` will be calling all notification so I gave it a user with such config that would allow them to receive all welcome notifications (currently only 2). This is basically along the line of `it "works"`